DittoHeartbeat/src/main/java/live.ditto.dittoheartbeat/Heartbeat.kt

```diff
@@ -0,0 +1,149 @@
+package live.ditto.dittoheartbeat
+
+import live.ditto.Ditto
+import android.os.Handler
+import android.os.Looper
+import androidx.compose.material3.Text
+import live.ditto.DittoConnectionType
+import live.ditto.DittoPeer
+import live.ditto.DittoPresenceObserver
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+data class HeartbeatConfig(
+    val interval: Long,
+    val collectionName: String,
+    val metaData: Map<String, Any>
+)
+
+data class HeartbeatInfo(
+    val id: Map<String, String>,
+    val lastUpdated: String,
+    val presence: Presence?,
+)
+
+data class Presence(
+    val totalConnections: Int,
+    val connections: List<DittoPeer>,
+)
+fun startHeartbeat(ditto: Ditto, config: HeartbeatConfig, onDataLog: (HeartbeatInfo) -> Unit): Handler {
```

Comment:
What do you think about returning a flow?
```
fun heartbeatFlow(
   ditto: Ditto,
   delayTimeMs Int = 30_000,
   ...
) = flow {
  while(isActive) {
      ...
      val info = HeartbeatInfo(...)
      emit(info)
      delay(delayTimeMs)
  }
}
```

Having a flow gives you the benefits of free cancellation once the flow is not collected.
Additionally because flows are cold, nothing is executed until it is observed.